e2e-tests/test/evm-apis.test.ts

```diff
@@ -56,13 +56,13 @@ describe("evm_setNextBlockTimestamp", function () {
     const userWallet = Wallet.createRandom().connect(provider);
 
     // Act
-    await provider.send("evm_setNextBlockTimestamp", [expectedTimestamp]);
+    await provider.send("evm_setNextBlockTimestamp", [expectedTimestamp.toString(16)]);
 
     await wallet.sendTransaction({
       to: userWallet.address,
       value: ethers.utils.parseEther("0.1"),
     });
-    expectedTimestamp += 2; // New transaction will add two blocks
+    expectedTimestamp += 1; // New transaction will add two blocks
```

Comment:
```suggestion
    expectedTimestamp += 1; // New transaction will add one block
```

Reply:
No, it's 2 blocks: the first is the original `expectedTimestamp` (the one used as an argument to `evm_setNextBlockTimestamp`, because that's the next block after calling it), the second has timestamp 1 second larger than that.

Reply:
you changed `expectedTimestamp += 2;` to `expectedTimestamp += 1;`
why then the comment is ` New transaction will add two blocks`

Reply:
Because the second of 2 blocks will have timestamp exactly 1 larger than the argument to call `evm_setNextBlockTimestamp`. The number of generated blocks did not change - only their timestamps.

Reply:
i can not understand why 1 tx adds 2 blocks and not 1

Reply:
Hmm, I didn't give that much thought before, but apparently `Wallet.sendTransaction` calls `eth_estimateGas` to determine the fee for the transaction, which generates the first block. Now, that block apparently isn't added to the node state, so it's debatable whether it _should_ increase the timestamp - but it does...

Reply:
i think it is not correct behavior

Reply:
> i think it is not correct behavior

Well, first of all I'd argue it's out of scope for this PR (and so might get an issue of its own), but second of all, why is it incorrect? There is no expectation of the timestamp staying fixed in normal usage - a normal user cannot ignore all the other users producing other transactions... The only expectation is that the timestamp only goes up...

Reply:
I see. The thing is that we increase the current timestamp right after tx is executed.

Reply:
Well, yes, but the real reason is that we have a block for every transaction (IOW 1-to-1 correspondence between blocks and transactions), and every block needs a different (larger) timestamp.